src/Core/Services/Implementations/CipherService.cs

```diff
@@ -646,10 +646,19 @@ public async Task DeleteFolderAsync(Folder folder)
             }
         }
 
+        var existingFolders = new List<Folder>();
         // Init. ids for folders
         foreach (var folder in folders)
         {
-            folder.SetNewId();
+            if (folder.Id == Guid.Empty || (await _folderRepository.GetByIdAsync(folder.Id))?.UserId != userId)
```

Comment:
this sql call should be batched to gather all invalid folders rather than making N sql calls